service/src/main/kotlin/io/provenance/api/frameworks/objectStore/ObjectStoreService.kt

```diff
@@ -1,57 +1,82 @@
-    ): ByteArray {
-        val future = client.get(hash, publicKey)
-        val res: DIMEInputStream = future.get(osConfig.timeoutMs, TimeUnit.MILLISECONDS)
-        return res.getDecryptedPayload(DirectKeyRef(publicKey, privateKey)).readAllBytes()
-    }
-
-    override fun <T> store(client: OsClient, message: T, publicKey: PublicKey, additionalAudiences: Set<PublicKey>): StoreProtoResponse {
-        val future = when (message) {
-            is InputStream -> client.put(
-                message,
-                publicKey,
-                Pen(ProvenanceKeyGenerator.generateKeyPair(publicKey)),
-                message.available().toLong(),
-                additionalAudiences
-            )
-            is Message -> client.put(
-                message,
-                publicKey,
-                Pen(ProvenanceKeyGenerator.generateKeyPair(publicKey)),
-                additionalAudiences
-            )
-            else -> {
-                throw IllegalArgumentException("Not supported file type to store against EOS!")
+    ): ByteArray =
+        when (client) {
+            is OsClient -> {
+                val future = client.get(hash, publicKey)
+                val res: DIMEInputStream = future.get(osConfig.timeoutMs, TimeUnit.MILLISECONDS)
+                res.getDecryptedPayload(DirectKeyRef(publicKey, privateKey)).readAllBytes()
+            }
+            is GatewayClient -> {
+                client.getObject(
+                    hash.toString(),
+                    GatewayJwt.KeyPairJwt(KeyPair(publicKey, privateKey)),
+                    Duration.ofMillis(osConfig.timeoutMs)
+                ).`object`.objectBytes.toByteArray()
             }
+            else -> throw IllegalArgumentException("Unsupported client type while retrieving object!")
         }
 
-        return future.get(osConfig.timeoutMs, TimeUnit.MILLISECONDS).toModel()
-    }
+    override fun <T> store(client: T, message: ByteArray, publicKey: PublicKey, privateKey: PrivateKey, additionalAudiences: Set<PublicKey>, type: String?): StoreProtoResponse =
+        when (client) {
+            is OsClient -> {
+                client.put(
+                    ByteArrayInputStream(message),
+                    publicKey,
+                    Pen(ProvenanceKeyGenerator.generateKeyPair(publicKey)),
+                    message.size.toLong(),
+                    additionalAudiences,
+                    type?.let { mapOf("type" to type) } ?: mapOf()
+                ).get(osConfig.timeoutMs, TimeUnit.MILLISECONDS).toModel()
+            }
+
+            is GatewayClient -> {
+//                client.adminPutDataStorageAccount(
+//                    "tp1s2c62ke0mmwhqxguf7e2pt6e98yq38m4atwhwl",
+//                    true,
+//                    Duration.ofMillis(10000)
+//                ).curl
+                    val jwt = GatewayJwt.KeyPairJwt(KeyPair(publicKey, privateKey)).createJwt(false, OffsetDateTime.now().plusDays(1))
+                println(jwt)
```

Comment:
also :sus: if this prints a private key, plz remove

Reply:
this was testing. removed.